### PR TITLE
make sure that 'before' event emitted from Request only once

### PR DESCRIPTION
**Pull request by darkdarkdragon**

### CODE REVIEW

**Comment by wltsmrz** [comment]
LGTM

File: src/core/request.js [hunk 1]
```diff
@@ -44,13 +44,15 @@ Request.prototype.request = function(servers, callback_) {
   const callback = typeof servers === 'function' ? servers : callback_;
 
   this.emit('before');
+
+  const wasRequested = this.requested;
```
Comment (clark800) [inline src/core/request.js]: Is there a reason we have to call `callback` even if this was already requested? Could we just throw an exception if request gets called twice?

Comment (darkdarkdragon) [inline src/core/request.js]: Maybe user wants another callback added? From reading previous code I've suggested that logic.

Comment (clark800) [inline src/core/request.js]: Seems like unnecessary complexity unless there is a good use case, in which case we should add a comment to explain.

Comment (darkdarkdragon) [inline src/core/request.js]: I've made changes that throws error in case of request called twice
https://github.com/ripple/ripple-lib/pull/572

**Pull request merged**